x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/BlockHash.java

```diff
@@ -46,6 +50,19 @@ public abstract sealed class BlockHash implements Releasable, SeenGroupIds //
      */
     public abstract void add(Page page, GroupingAggregatorFunction.AddInput addInput);
 
+    /**
+     * Lookup all values for the "group by" columns in the page to the hash and return an
+     * {@link Iterator} of the values. The sum of {@link IntBlock#getPositionCount} for
+     * all blocks returned by the iterator will equal {@link Page#getPositionCount} but
+     * will "target" a size of {@code targetBlockSize}.
+     * <p>
+     *     Calling this will either {@link Page#releaseBlocks() release} the blocks immediately
```

Comment:
I find it a bit confusing that the lookup API releases the input Page itself, although the Javadoc explains this clearly. Should the caller manage the lifecycle of the input page instead? However, I'm totally fine if we decide to stick with this approach.